llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -7100,6 +7112,15 @@ def : Pat<(v4f16 (vector_insert (v4f16 V64:$Rn),
               (i64 0)),
             dsub)>;
 
+def : Pat<(v8i8 (vector_insert (v8i8 (undef)),
+            (i32 (vector_extract v8i16:$Vt, VectorIndexH:$idx)), (i64 0))),
+          (v8i8 (EXTRACT_SUBREG
+            (INSvi16lane
+              (v8i16 (IMPLICIT_DEF)),
+              (i64 0),
+              $Vt,
+              (VectorIndexH:$idx)), dsub))>;
```

Comment:
The remaining folds in this file are attempts to avoid regressions (though folds are still missing, and there are still regressions in this patch that did not exist in the original patch).

Reply:
I think tweaking the DAG combine could avoid the need for this